tests/qnn/test_keras.py

```diff
@@ -418,6 +418,25 @@ def c_default(w1, w2, w3, w4, w5, w6, w7, inputs=None):
         assert layer_out.shape == (batch_size, output_dim)
         assert np.allclose(layer_out[0], c(x[0], *weights))
 
+    @pytest.mark.parametrize("n_qubits, output_dim", indices_up_to(1))
+    @pytest.mark.parametrize("batch_size", [2])
+    @pytest.mark.parametrize("middle_dim", [5])
+    def test_call_broadcast(self, get_circuit, output_dim, middle_dim, batch_size, n_qubits):
+        """Test if the call() method performs correctly when the inputs argument is an arbitrary shape (that can
+        correctly be broadcast over), i.e., for input of shape (batch_size, dn, ... , d0) it outputs with shape
+        (batch_size, dn, ... , d1, output_dim)"""
+        c, w = get_circuit
+
+
+        layer = KerasLayer(c, w, output_dim)
+        x = tf.ones((batch_size, middle_dim, n_qubits))
+
+        layer_out = layer(x)
+
```

Comment:
Either in this test or a separate one, it would be good to check if the gradient is still accessible (e.g., even if we don't know the expected value, just to check it can be calculated and is not `None`).

Reply:
(The same suggestion applies to the torch test)

Reply:
Good idea, have done that now